File: Array/11. Container With Most Water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        lo=0
        hi = len(height)-1
        maxArea = 0
        while lo<hi:
            h = height[lo] if height[lo]<height[hi] else height[hi]
            Area = h*(hi-lo)
            maxArea = Area if Area>maxArea else maxArea
            if height[lo]<height[hi]:
                lo += 1
            else:
                hi -= 1
        return maxArea

File: Array/11. Container With Most Water.java
class Solution {
    public int maxArea(int[] height) {
        int lo = 0;
        int hi = height.length-1;
        int maxArea=0;
        while(lo<hi){
            int h = (height[lo]<height[hi]) ? height[lo]:height[hi];
            int Area = h*(hi-lo);
            maxArea = (Area>maxArea) ? Area:maxArea;
            if(height[lo]<height[hi]){
                lo++;
            }else{
                hi--;
            }
        }
        return maxArea;
    }
}